tests/unit/test_server_connection.py

```diff
@@ -86,7 +86,7 @@ def test_run_query_exceptions(mock_server_connection, caplog):
     ):
         with pytest.raises(Exception, match="fake exception"):
             with caplog.at_level(logging.ERROR):
-                mock_server_connection.run_query("fake query")
+                mock_server_connection.run_query("fake query", log_on_exception=True)
```

Comment:
1) Can we enable it for all tests we have?
2) How does a user enable log on exception? `collect(log_on_exception=True)` ?

Reply:
added these